### PR TITLE
[TS-6493] add support for rack-aware

### PR DESCRIPTION
**Pull request by nitzanharel**

### CODE REVIEW

**Review by TC-aLi — APPROVED** [review]

**Comment by TT-lwang** [comment]
It seems marina has very strick check on code format, which makes the build fail

File: src/marina_pool_server.erl [hunk 1]
```diff
@@ -131,30 +136,35 @@ connect(Ip, Port) ->
...
-filter_datacenter([[RpcAddress, _Datacenter, Tokens] | T], undefined) ->
-    [{RpcAddress, Tokens} | filter_datacenter(T, undefined)];
-filter_datacenter([[RpcAddress, Datacenter, Tokens] | T], Datacenter) ->
-    [{RpcAddress, Tokens} | filter_datacenter(T, Datacenter)];
-filter_datacenter([_ | T], Datacenter) ->
-    filter_datacenter(T, Datacenter).
-
-nodes([], _Port) ->
+filter_datacenter_and_racks([[RpcAddress, Datacenter, Rack, Tokens] | T], Retrieved_Datacenter, Configured_Racks)
+    when Retrieved_Datacenter == undefied; Retrieved_Datacenter == Datacenter->
```
Comment (zvoykish) [inline src/marina_pool_server.erl]: typo undefied

Comment (zvoykish) [inline src/marina_pool_server.erl]: Also, what's the use case for undefined data center?

File: rebar.config [hunk 2]
```diff
@@ -10,11 +10,11 @@
   {foil, ".*",
     {git, "https://github.com/lpgauth/foil.git", {tag, "0.1.1"}}},
   {lz4, ".*",
-    {git, "git@github.com:tigertext/erlang-lz4.git", {branch, "develop"}}},
+    {git, "https://github.com/tigertext/erlang-lz4.git", {branch, "develop"}}},
```
Comment (zvoykish) [inline rebar.config:13]: curious why git->https...?

**Review by zvoykish — CHANGES_REQUESTED** [review]

File: src/marina_pool_server.erl (hunk 1 above)
Comment (nitzanharel) [reply src/marina_pool_server.erl]: fixed

Comment (nitzanharel) [reply src/marina_pool_server.erl]: undefined data center is where there no cluster. e.g. - localhost and one node.

**Pull request merged**